day1/adventcode2015_day1_refactor.py

```diff
@@ -0,0 +1,32 @@
+with open('input.txt', 'r') as floors:
+    visited_floors: str = floors.readline().strip()
+
+
+def count_by_character(input_string, counter=0):
+    match input_string:
+        case '(':
+            counter += 1
+        case ')':
+            counter -= 1
+    return counter
+
+
+def count_santa_floor_and_basement_index(floors_input: str) -> tuple[int, int or None]:
```

Comment:
I'd stick with tuple[int, int] and just raise an error whenever there is no 'basement_index" at the end